File: user/model/user_model/response.go
package user_model

type Response struct {
	Result  bool        `json:"result"`  // true: success, false: error
	Message string      `json:"message"` // error message
	Data    interface{} `json:"data"`    // response data
}

func SuccessResponse(data interface{}) Response {
	return Response{
		Result:  true,
		Message: "success",
		Data:    data,
	}
}

func ErrorResponse(message string) Response {
	return Response{
		Result:  false,
		Message: message,
		Data:    nil,
	}
}

File: order/order/Response.cs
using System.Text.Json.Serialization;

namespace order;

public class Response<T>
{
    //[JsonPropertyName("data")]
    public T Data { get; set; }
    //[JsonPropertyName("message")]
    public string Message { get; set; }
    //[JsonPropertyName("result")]
    public bool Result { get; set; }
}

public static class RequestResponse
{
    public static Response<T> BadRequest<T>(string message = "Unknown Error.")
    {
        return new Response<T>() { Message = message };
    }
    
    public static Response<T> NotFound<T>()
    {
        return new Response<T>() { Message = "Data Not Exist." };
    }
}